staging/src/k8s.io/apimachinery/pkg/runtime/serializer/versioning/versioning.go

```diff
@@ -166,9 +166,22 @@ func (c *codec) Decode(data []byte, defaultGVK *schema.GroupVersionKind, into ru
 // Encode ensures the provided object is output in the appropriate group and version, invoking
 // conversion if necessary. Unversioned objects (according to the ObjectTyper) are output as is.
 func (c *codec) Encode(obj runtime.Object, w io.Writer) error {
-	switch obj.(type) {
-	case *runtime.Unknown, runtime.Unstructured:
+	switch obj := obj.(type) {
+	case *runtime.Unknown:
 		return c.encoder.Encode(obj, w)
+	case runtime.Unstructured:
+		// avoid conversion roundtrip if GVK is the right one already or is empty (yes, this is a hack, but the old behaviour we rely on in kubectl)
```

Comment:
Got a list of where?  Not converting if the target is valid seems fine.  Not converting if the existing doesn't have a GVK.... How does that even happen for unstructured?

I can see a followup issue, but it's still weird.